prisma/schema.prisma

```diff
@@ -66,11 +66,12 @@ model UserReferral {
 }
 
 model UserReferralCode {
-  id     Int     @id @default(autoincrement())
-  userId Int
-  user   User    @relation(fields: [userId], references: [id], onDelete: Cascade)
-  code   String  @unique
-  note   String?
+  id        Int       @id @default(autoincrement())
+  userId    Int
+  user      User      @relation(fields: [userId], references: [id], onDelete: Cascade)
+  code      String    @unique
+  note      String?
+  deletedAt DateTime?
 
   referees UserReferral[]
```

Comment:
I learned from Koen that we can have an index on `userId` to make querying these faster 🤓 
```suggestion
  referees UserReferral[]
  
  @@index([userId], type: Hash)
```